src/ynEIGEN/TokenStakingNodesManager.sol

```diff
@@ -126,6 +134,14 @@ contract TokenStakingNodesManager is AccessControlUpgradeable, ITokenStakingNode
         maxNodeCount = init.maxNodeCount;
     }
 
+    function initializeV2(
+        address _redemptionAssetsVault,
+        address _withdrawer
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) reinitializer(2) notZeroAddress(_redemptionAssetsVault) {
```

Comment:
Given the experience with ynETH and the issues with this, we should not use

onlyRole(DEFAULT_ADMIN_ROLE)

and call the initialize as part of upgradeAndCall().

Need to think carefully through this but i believe it would be correct.

Reply:
done